### ISSUE TITLE
trans_default_domain does not work with named arguments

### ISSUE THREAD

**Issue by stof** [symfony/symfony#10859]
the TranslationDefaultDomainVisitor deos not deal with named arguments when altering the ``trans`` and ``transchoice`` filter nodes to insert the domain. This means that if you force the locale using a named argument, you will get an exception ``Positional arguments cannot be used after named arguments``

**Comment by stof** [comment]
```jinja
{% trans_default_domain 'test' %}
{{ 'foobar'|trans(locale='fr') }}
```

**Comment by stof** [comment]
an easy fix could be to always use a named argument to inject the domain (needs a check to ensure it works in all cases). Otherwise, a more complex fix would be needed

**Comment by aitboudad** [comment]
fixed in #13250
